plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md

```diff
@@ -0,0 +1,388 @@
+
+#### Options for `text` fields
+
+As well as the options above, text fields also support a `required` option. If this is `true` then the shopper _must_ provide a value for this field during the checkout process.
+
+| Option name     | Description                                                                                                                         | Required? | Example                                      | Default value |
+|-----------------|-------------------------------------------------------------------------------------------------------------------------------------|-----------|----------------------------------------------|---|
+| `required` | If this is `true` then the shopper _must_ provide a value for this field during the checkout process. | No | `true` | `false` |
+
+#### Options for `select` fields
+
+As well as the options above, select fields must also be registered with an  `options` option. This is used to specify what options the shopper can select.
+
+The `optionalLabel` option will never be shown as select fields are _always_ required.
+
+<table>
+	<tr>
+		<th>Option name</th>
+		<th>Description</th>
+		<th>Required?</th>
+		<th>Example</th>
+		<th>Default value</th>
+	</tr>
+	<tr>
+		<td>
+
+`options`
+
+</td>
+		<td>
+
+An array of options to show in the select input. Each options must be an array containing a `label` and `value` property. Each entry must have a unique `value`. Any duplicate options will be removed. The `value` is what gets submitted to the server during checkout and the `label` is simply a user-friendly representation of this value. It is not transmitted to the server in any way.</td>
+<td>Yes</td>
+<td>
+
+	[
+		[
+			'value' => 'store_1',
+			'label' => 'Our London Store'
+		],
+		[
+			'value' => 'store_2',
+			'label' => 'Our Paris Store'
+		],
+		[
+			'value' => 'store_3',
+			'label' => 'Our New York Store'
+		]
+	]
+
+</td>
+		<td>
+			No default - this must be provided.
+		</td>
+	</tr>
+</table>
+
+#### Options for `checkbox` fields
+
+The checkbox field type does not have any specific options, however `required` will always be `false` for a checkbox field. Making a checkbox field required is not supported.
+
+### Attributes
+
+Adding additional attributes to checkbox and text fields is supported. Adding them to select fields is **not**.
+
+These attributes have a 1:1 mapping to the HTML attributes on `input` elements (except `pattern` on checkbox).
+
+The supported attributes are:
+
+- `data-*` attributes
+- `aria-*` attributes
+- `autocomplete`
+- `autocapitalize`
+- `pattern` (not supported on checkbox fields)
+- `title`
+- `maxLength` (equivalent to `maxlength` HTML attribute)
+- `readOnly` (equivalent to `readonly` HTML attribute)
+
+`maxLength` and `readOnly` are in camelCase because the attributes are rendered on a React element which must receive them in this format.
+
```

Comment:
We should mention that we omitted certain attributes intentionally but are welcome to hear feedback, those attributes are `autofocus`, `disabled`, `readonly`.